streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobSet.scala

```diff
@@ -61,13 +61,14 @@ case class JobSet(
     processingEndTime - time.milliseconds
   }
 
-  def toBatchInfo: BatchInfo = {
+  def toBatchInfo(errorMessage: Option[String] = None): BatchInfo = {
```

Comment:
I would prefer setting the error message in the JobSet, and then keeping the to BatchInfo() simple. In that way JobSet stays a strict superset of the all the information available in BatchInfo.